lib/BridgeManager.php

```diff
@@ -0,0 +1,399 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk;
+
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IL10N;
+use OCP\Files\IAppData;
+use OCP\Files\NotFoundException;
+use OCP\Files\SimpleFS\ISimpleFolder;
+use OCP\IURLGenerator;
+
+use OCA\Talk\Exceptions\ImpossibleToKillException;
+
+class BridgeManager {
+	public const EVENT_TOKEN_GENERATE = self::class . '::generateNewToken';
+
+	/** @var IDBConnection */
+	private $db;
+	/** @var IConfig */
+	private $config;
+	/** @var IAppData */
+	private $appData;
+	/** @var IL10N */
+	private $l;
+
+	public function __construct(IDBConnection $db,
+								IConfig $config,
+								IAppData $appData,
+								IURLGenerator $urlGenerator,
+								Manager $manager,
+								IL10N $l) {
+		$this->db = $db;
+		$this->config = $config;
+		$this->urlGenerator = $urlGenerator;
+		$this->appData = $appData;
+		$this->manager = $manager;
+		$this->l = $l;
+	}
+
+	public function getBridgeOfRoom(string $token): array {
+		$defaultParts = sprintf(
+			'{"enabled":false,"pid":0,"parts":[{"type":"nctalk","server":"","login":"","password":"","channel":"%s"}]}',
+			$token
+		);
+		$bridgeJSON = $this->config->getAppValue('spreed', 'bridge_' . $token, $defaultParts);
+		$bridge = json_decode($bridgeJSON, true);
+		return $bridge;
+	}
+
+	public function editBridgeOfRoom(string $token, bool $enabled, array $parts = []): bool {
+		$currentBridge = $this->getBridgeOfRoom($token);
+		$newBridge = [
+			'enabled' => $enabled,
+			'pid' => isset($currentBridge['pid']) ? $currentBridge['pid'] : 0,
+		];
+		if (count($parts) > 0) {
+			$newBridge['parts'] = $parts;
+		} else {
+			$newBridge['parts'] = $currentBridge['parts'];
+		}
+
+		// edit/update the config file
+		$this->editBridgeConfig($token, $newBridge);
+
+		// check state and manage the binary
+		$pid = $this->checkBridgeProcess($token, $newBridge);
+		$newBridge['pid'] = $pid;
+
+		// save config
+		$newBridgeJSON = json_encode($newBridge);
+		$this->config->setAppValue('spreed', 'bridge_' . $token, $newBridgeJSON);
+
+		return true;
+	}
+
+	public function deleteBridgeOfRoom(string $token): bool {
+		// first potentially kill the process
+		$currentBridge = $this->getBridgeOfRoom($token);
+		$currentBridge['enabled'] = false;
+		$this->checkBridgeProcess($token, $currentBridge);
+		// then actually delete the config
+		$bridgeJSON = $this->config->deleteAppValue('spreed', 'bridge_' . $token);
+		return true;
+	}
+
+	public function checkAllBridges() {
+		// TODO call this from time to time to make sure everything is running fine
+		$this->manager->forAllRooms(function ($room) {
+			$token = $room->getToken();
+			if ($room->getType() === Room::GROUP_CALL or $room->getType() === Room::PUBLIC_CALL) {
```

Comment:
```suggestion
			if ($room->getType() === Room::GROUP_CALL || $room->getType() === Room::PUBLIC_CALL) {
```

Never use `and` and `or` in php, it always executes all conditions. You had an other occurance with `isset() and ...` which will still trigger an error because of using undefined.

Reply:
Good to know. Thanks